torch/utils/_mode_utils.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from typing import Iterator
+
+
+# a class for the helper function to package all the info about the Mode class
+# so the helper function can access it without needing a circular import
+class _ModeInfo:
```

Comment:
Is there a design for ModeInfo anywhere? I suppose I'll find out more about what it's used for as I read this PR, but the comment here doesn't really help (and we didn't have anything like ModeInfo for torch function)

Reply:
Suggestion: make this a dataclass

Reply:
done!